source/server/server.cc

```diff
@@ -190,11 +190,15 @@ MetricSnapshotImpl::MetricSnapshotImpl(Stats::Store& store, TimeSource& time_sou
         gauges_.push_back(gauge);
       });
 
-  snapped_histograms_ = store.histograms();
-  histograms_.reserve(snapped_histograms_.size());
-  for (const auto& histogram : snapped_histograms_) {
-    histograms_.push_back(*histogram);
-  }
+  store.forEachSinkedHistogram(
```

Comment:
can we verify what happens for histograms that are never sinked but where we call recordValue regularly?

Do they consume unbounded memory? Maybe make a unit test that runs an infinite loop with a bunch of recordValue() and see if it consumes increasing memory while you let it spin.

If so, then this might be a memory leak.

This case is also why I wanted to do histograms separate from the other stuff, as we'll need to figure out how to deal with this.